vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -316,11 +301,10 @@ import { ComboBoxPlaceholder } from '@vaadin/combo-box/src/vaadin-combo-box-plac
         // The filter used is the one from combobox, not the lastFilter stored since
         // that may not reflect user's input.
         const performClientSideFilter = tryCatchWrapper(function (page, callback) {
-          const dataProviderMixin = getDataProviderMixin();
           let filteredItems = page;
 
-          if (dataProviderMixin.filter) {
-            filteredItems = page.filter((item) => comboBox.$connector.filter(item, dataProviderMixin.filter));
+          if (comboBox.filter) {
+            filteredItems = page.filter((item) => comboBox.$connector.filter(item, comboBox.filter));
```

Comment:
This change brings up a problem that the filter property is synchronized **after** the data provider call, which causes the filtered results to lag one character behind:

https://user-images.githubusercontent.com/357820/173076458-6779f213-44f1-4aa2-9b62-02736668457a.mp4

I had the same issue initially when adapting the connector, and had a solution where I passed the actual filter from the data provider callback into the `performClientSideFilter` call. I reverted that because it was not necessary anymore after using the data provider element directly. So maybe can try something like that as an alternative here.

Reply:
Done.

Reply:
Seems that the change broke some other tests.

Reply:
This is fixed.